librosa/decompose.py

```diff
@@ -322,7 +357,7 @@ def hpss(S, kernel_size=31, power=2.0, mask=False):
         perc[zero_harm & zero_perc] = 0.5
 
         # Compute harmonic mask
-        mask_harm = harm / (harm + perc)
-        mask_perc = perc / (harm + perc)
+        mask_harm = (harm >  perc * margin_harm).astype(float)
```

Comment:
I need to fix this. 
1. Complete test coverage
2. Eliminate effects.residual, doesn't makes sense.
3. **kwargs is a good idea.
4. Fix soft masks 

I also implemented an alternative noise/sinusoidal separation using phase acceleration thresholds based on [[1]](http://vud.org/papers/ApelACMC2014.pdf). Do you want me to add this to librosa too?

This was part of my larger experiment: a phase vocoder time stretch that treats noise, harmony, and percussion components differently. It seems:
 - *Percussion* needs to be unstretched, phases unchanged.
 - *Harmony/sinusoidal* just needs to respect ∆phase
 - *Noise* The more it stretches , the more it becomes sinusoidal, which produces artifacts. The noise synthesis method in the paper (white noise filtered by smoothed magnitude spectrum) is OK, it avoids the artifacts, but the fidelity of texture is lost. I could try statistical texture methods next. Thoughts?